python/shotgun_desktop/upgrade_startup.py

```diff
@@ -48,49 +45,30 @@ def upgrade_startup(splash, sgtk, app_bootstrap):
     if not _supports_get_from_location_and_paths(sgtk):
         return False
 
-    current_desc = sgtk.deploy.descriptor.get_from_location_and_paths(
-        sgtk.deploy.descriptor.AppDescriptor.FRAMEWORK,
-        app_bootstrap.get_shotgun_desktop_cache_location(),
-        os.path.join(
-            app_bootstrap.get_shotgun_desktop_cache_location(), "install"
-        ),
+    current_desc = sgtk.descriptor.create_descriptor(
+        sgtk.get_authenticated_user().create_sg_connection(),
+        sgtk.descriptor.Descriptor.FRAMEWORK,
         get_location(sgtk, app_bootstrap)
     )
 
     # A Dev descriptor means there is nothing to update. Early out so that we don't show
     # "Getting Shotgun Desktop updates...""
-    if isinstance(current_desc, sgtk.deploy.dev_descriptor.TankDevDescriptor):
+    if current_desc.is_dev():
         logger.info("Desktop startup using a dev descriptor, skipping update...")
         return False
 
     splash.set_message("Getting Shotgun Desktop updates...")
     logger.info("Getting Shotgun Desktop updates...")
 
-    # Some clients block tank.shotgunstudio.com with a proxy. Normally, this isn't an issue
-    # because those clients will be using a locked site config, which won't try to connect
-    # to tank.shotgunstudio.com. The Desktop startup update code however always phones home.
-    # Beacuse of this, we'll try to find the latest version but accept that it may fail.
-
-    # Local import because sgtk can't be imported globally in the desktop startup.
-    from tank_vendor.shotgun_api3.lib import httplib2
+    if not current_desc.has_remote_accesss():
+        logger.warning("Could not update %r: remote access not available.", current_desc)
```

Comment:
They are in the logs regardless. However, I've moved the Getting Shotgun updates after the remote test so that clients who are not connected to the internet don't see this message